analyzers/tests/SonarAnalyzer.UnitTest/Extensions/BlockSyntaxExtensionsTests.cs

```diff
@@ -0,0 +1,316 @@
+
+        [TestMethod]
+        public void IsEmpty_BlockMethodWithComment_Singleline_Off()
+        {
+            var block = @"
+{
+    // Single line
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            declaration.Body.Should().BeEmpty(treatCommentsAsContent: false);
+        }
+
+        [TestMethod]
+        public void IsEmpty_BlockMethodWithComment_Empty_On()
+        {
+            var block = @"
+{
+    //
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            declaration.Body.Should().BeNotEmpty(treatCommentsAsContent: true); // This is a questionable behavior.
+        }
+
+        [TestMethod]
+        public void IsEmpty_BlockMethodWithComment_Multiline_On()
+        {
+            var block = @"
+{
+    /* Multi line */
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            declaration.Body.Should().BeNotEmpty(treatCommentsAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_BlockMethodWithComment_Multiline_Off()
+        {
+            var block = @"
+{
+    /* Multi line */
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            declaration.Body.Should().BeEmpty(treatCommentsAsContent: false);
+        }
+
+        [TestMethod]
+        public void IsEmpty_BlockMethodWithConditionalCompilation_WithDisabledCode_On()
+        {
+            var block = @"
+{
+#if SomeThing
+    var i = 0;
+#endif
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            declaration.Body.Should().BeNotEmpty(treatConditionalCompilationAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_BlockMethodWithConditionalCompilation_WithDisabledCode_Off()
+        {
+            var block = @"
+{
+#if SomeThing
+    var i = 0;
+#endif
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            declaration.Body.Should().BeEmpty(treatConditionalCompilationAsContent: false);
+        }
+
+        [TestMethod]
+        public void IsEmpty_BlockMethodWithConditionalCompilation_WithEmptyRegion_On()
+        {
+            var block = @"
+{
+#region SomeRegion
+#endregion
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            declaration.Body.Should().BeEmpty(treatConditionalCompilationAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_BlockMethodWithConditionalCompilation_WithEmptyConditional_On()
+        {
+            var block = @"
+{
+#if SomeCondition
+#endif
+}
+";
+            var declaration = MethodDeclarationForBlock(block);
+            declaration.Body.Should().BeEmpty(treatConditionalCompilationAsContent: true);
+        }
+
+        [DataTestMethod]
+        [DataRow(true, true, false)]
+        [DataRow(false, true, false)] // isEmpty should be true here, because the content of the conditional is just a comment and a comment should not be treated as content.
```

Comment:
>  Conditional compilation is considered content

That is true with one nuance: It needs to be NonEmpty too (there is one test case documenting this). But because it only contains comments, it should still be considered empty.